### PR TITLE
Weighted loss for classification and Cox model.

### PR DESCRIPTION
**Pull request by xyang23**
1. Weighted loss for classification: 
- add weights for different classes as a parameter in the criterion function
- use mice protein dataset an example
2. Cox model:
- add LassoNetCoxRegressor for survival prediction
- use HNSCC dataset as an example
- add plot_cox_loss in utils to plot the results

### CODE REVIEW

**Comment by louisabraham** [comment]
Hi, I made a few changes. Mostly cleaned your code.

- moved functions relative to survival to cox.py
- removed the RegressorMixin and MultiOutputMixin types from LassoNetCoxRegressor
- removed plot_cox_loss
- implemented my own O(n log n) concordance-index
- replaced `(log_cse * events).nonzero()` with `events.nonzero()`, in your Cox loss. I don't understand what the value of a `logcumsumexp` being 0 would change.
- renamed `weighted_loss` as `class_weight` to match [sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html)
- added a check in `_init_model` to force the observed number of classes to be the same as the number of class weights.
- replaced `miceprotein_weighted.py` by simply adding it to `miceprotein.py`, thus avoiding a lot of code duplication
- renamed `HNSCC.py` as `cox_regression.py`

and a few other things...

**Comment by louisabraham** [comment]
Could you add some README in HNSCC_data to explain how that data is made?
